### ISSUE TITLE
Sometimes headers are lowercased

### ISSUE THREAD

**Issue by emilwojtaszek** [pointfreeco/swift-web#185]
Sometimes by basic authentication fails, because the authorization header is typed lowercased (mostly on Linux environment) and it can't be found. Source of the problem is probably somewhere in NIO framework, but it can be easily fixed by forcing lowercase like here:

```swift
  let auth = request.allHTTPHeaderFields?.first(where: { $0.key.lowercased() == "authorization" })?.value ?? ""
```

This line is over here:
https://github.com/pointfreeco/swift-web/blob/master/Sources/HttpPipeline/SharedMiddlewareTransformers.swift#L163

**Comment by stephencelis** [comment]
Sounds good to us! Would you like to PR the fix or should we make the change?

**Comment by emilwojtaszek** [comment]
you can find PR here https://github.com/pointfreeco/swift-web/pull/186

but not sure if this is the correct approach... maybe better is to create some middleware which just rewrites URLRequest headers?

**Comment by emilwojtaszek** [comment]
also, not sure if I'm able to write proper tests for that case, please check this out https://github.com/pointfreeco/swift-web/pull/186/files#diff-42876cbbcd6c88aa843a9519adeb0c5aR2 even if I provide lowercased header name, URLRequest automatically capitalize this, guessing there is no similar behaviour on Linux?

**Comment by stephencelis** [comment]
Fixed by #186.
